src/catch2/internal/catch_clara.hpp

```diff
@@ -448,16 +469,21 @@ namespace Catch {
                     m_ref( ref ) {}
 
             public:
-                template <typename T>
+                template <typename T, typename = typename std::enable_if<!Detail::is_unary_function<T>::value>::type>
                 ParserRefImpl( T& ref, std::string const& hint ):
                     m_ref( std::make_shared<BoundValueRef<T>>( ref ) ),
                     m_hint( hint ) {}
 
-                template <typename LambdaT>
+                template <typename LambdaT, typename = typename std::enable_if<Detail::is_unary_function<LambdaT>::value>::type>
                 ParserRefImpl( LambdaT const& ref, std::string const& hint ):
                     m_ref( std::make_shared<BoundLambda<LambdaT>>( ref ) ),
                     m_hint( hint ) {}
 
+                template <typename LambdaT>
+                ParserRefImpl( accept_many_t, LambdaT const& ref, std::string const& hint ):
```

Comment:
If the "accept many" tag is placed as the last argument, couldn't we do without the sfinae?

Reply:
I don't think so. The reason for the sfinae is not the added `accept_many_t` feature, but the ability to pass named lambdas, i.e.
```c++
auto foo = [](std::string opt) { ... };

clara::Opt(foo, "foo")["--foo"]
```

Before, the passed lambda had to be "not convertible to non-const ref".

Reply:
Ok, makes sense. About as much as C++ tmp can make sense :smiley: